### PR TITLE
Returning chatroom users-list on /list command --updated with master

### PR DESCRIPTION
**Pull request by Vivek-blip**

### CODE REVIEW

**Comment by Vivek-blip** [comment]
I have updated with master 👍

**Comment by t0xic0der** [comment]
@nat236919 Could you please do a review to check if the code hails from the most recent commit to `master` or not?

Once done, I'd get into reviewing the functionality.

File: cnew.py [hunk 1]
```diff
@@ -49,9 +51,12 @@ async def producer_handler(cphrsuit, websocket, username, chatroom, servaddr):
         footelem = HTML("<b>[" + chatroom + "]</b>" + " <b>" + username.strip() + "</b> > End-to-end encryption enabled on '" + servaddr + "' - Hit Ctrl+C to EXIT")
         while True:
             with patch_stdout():
-                mesgtext = await sess.prompt_async(lambda:"[" + obtntime() + "] " + formusnm(str(username)) + " > ", bottom_toolbar=footelem, validator=emtyfind(), refresh_interval=0.5, prompt_continuation=lambda width, line_number, is_soft_wrap: " " * width)
-            senddata = json.dumps({"username": username.strip(), "chatroom": chatroom, "mesgtext": mesgtext.strip()})
-            senddata = cphrsuit.encrjson(senddata)
+                mesgtext = await sess.prompt_async("[" + obtntime() + "] " + formusnm(str(username)) + " > ", bottom_toolbar=footelem, validator=emtyfind())
```
Comment (nat236919) [inline cnew.py:54]: This line is missing **refresh_interval** and **prompt_continuation**

It should be
`mesgtext = await sess.prompt_async(lambda:"[" + obtntime() + "] " + formusnm(str(username)) + " > ", bottom_toolbar=footelem, validator=emtyfind(), refresh_interval=0.5, prompt_continuation=lambda width, line_number, is_soft_wrap: " " * width)`

File: main.py [hunk 2]
```diff
@@ -58,8 +62,11 @@ async def chatroom(websocket, path):
                     print_formatted_text(HTML("[" + obtntime() + "] " + "<b>USERJOINED</b> > <green>" + mesgjson.split(sepr)[0] + "@" + mesgjson.split(sepr)[1] + "</green>"))
                     await notify_mesej("SNCTRYZERO" + sepr + "USERJOINED" + sepr + mesgjson.split(sepr)[0] + sepr + mesgjson.split(sepr)[1] + sepr + str(getallus(mesgjson.split(sepr)[1])))
             else:
-                print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > " + helper_display.wrap_text(str(mesgjson))))
-                await notify_mesej(mesgjson)
+                if str(mesgjson) == "/list":
+                   await send_chatroommembers_list(websocket)
+                else:
+                    print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > " + str(mesgjson)))
```
Comment (nat236919) [inline main.py:68]: This one is missing **helper_display.wrap_text()**

It should be
`print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > " + helper_display.wrap_text(str(mesgjson))))`

**Review by nat236919 — CHANGES_REQUESTED** [review]
@t0xic0der @Vivek-blip 

As far as I can see, there are some missing features.

File: cnew.py (hunk 1 above)
Comment (t0xic0der) [reply cnew.py:54]: Yes, the line misses the parts of unfreezing time and live text wrapping.

File: main.py (hunk 2 above)
Comment (Vivek-blip) [reply main.py:68]: Oh my bad. I'll get it fixed

**Comment by Vivek-blip** [comment]
I have updated those lines 👍

**Review by nat236919 — APPROVED** [review]
@Vivek-blip Thank you.
@t0xic0der  LGTM

**Comment by Vivek-blip** [comment]
> @Vivek-blip Thank you.

@nat236919  thanks for the review 😊

**Comment by t0xic0der** [comment]
> @Vivek-blip Thank you.
> @t0xic0der LGTM

I can get into reviewing the functionality now. :smile:

**Comment by t0xic0der** [comment]
Looks good to me. @Vivek-blip has done a great job on this one.

I will make some QOL increments to the changes you made, so it would look like this.

- Basically, now that we have a `/list` functionality - we no longer need to list the connected users whenever any user joins or leaves.
- Also, just because the information is retrieved from the server (and not from any clients per se) - the sender name has been changed to SNCTRYZERO.

image

**Pull request merged**